allauth/socialaccount/models.py

```diff
@@ -338,6 +338,14 @@ def _lookup_by_email(self):
                 self.connect(context.request, address.user)
             else:
                 self.user = address.user
+            return
+
+        user = get_user_model().objects.filter(email__in=emails).first()
```

Comment:
Not all user models have an `email` field. Also, email casing needs to be taken into account, so this should likely be replaced by `filter_users_by_email()`. I think that the existing `EmailAddress.objects.lookup(emails).order_by("-verified", "user_id").first()` line could be replaced by it, so no need to duplicate the `if` that follows here.